### DISCUSSION TITLE
小幻老师你好，关于obsidian的md文件导入失败的咨询。

### DISCUSSION THREAD

**Discussion by wh000wh000** [Richasy/FantasyCopilot#48] [General]
![image](https://github.com/Richasy/FantasyCopilot/assets/16302138/aa317efa-746d-47a0-bd45-f8297745679b)
手边只有obsidian的文档具有md格式，所以想尝试一下一直以来都想尝试的借助GPT实现模仿自己的文风写作。 准备把知识库导入给它的时候出现了这个情况。似乎软件本身对obsidian的md格式不是特别支持？ 或者有什么其他原因吗？

**Comment by Richasy** [comment]
首先，你可以在设置页面找到日志记录文件夹，打开后选择当天的日志文件打开，里面有记录导入失败的原因。
如果是网络错误，请确保以下几件事：
1. 你的embedding模型名称填写正确
2. 如果你正在使用OpenAI，请确保你的账户注册时间超过了48小时（OpenAI对于API调用是有限额的，每分钟能解析的token数是有限的），你可以尝试先导入一个文本量较小的markdown试试

关于ob的markdown无法解析，这个应该不会。理论上只要你的文本文件没有加密，是纯文本，那就可以。
应用只是读取文本，解析是embedding模型做的

**Comment by wh000wh000** [comment]


**Reply by Richasy** [reply]
首先感谢你的支持，关于捐助，我现在还没有想好该怎么做，等我想好了，也许会开通捐助通道。但至少短期内，我希望使用这个项目的人能暂时不要有负担，尽情享受即可。

---

关于错误，日志中记载得很明确，因为你勾选了多个答案后，上下文的字符数已经超过了模型能接受的字符数上限，所以OpenAI直接拒绝了你的请求。

你可以尝试这个方法：
1. 将对话模型改为 `gpt-3.5-turbo-16k` （它的字符数上限是`gpt-3.5-turbo`的4倍）
2. 文本完成模型留空

这样可以缓解这个问题，但同时你也要注意不要勾选过多的上下文内容。

**Reply by wh000wh000** [reply]
好的 谢谢小幻老师

**Comment by wh000wh000** [comment]


**Reply by Richasy** [reply]
你可以先自己看一下log，主要是两个问题，第一个，key不对，第二个，模型名错了，是“text-embedding-ada-002”而不是"ext-embedding-ada-002"
